### ISSUE TITLE
Problem getting Organizr to observe me being out of network, uses local URLs.  

### ISSUE THREAD

**Issue by dgarra** [causefx/Organizr#1836]
###### Operating System: Docker *nix
<hr>

##### Problem Description:

Afternoon all, appreciate any direction on this. 

I have Organizr running next to my Nginx Proxy Manager in Docker inside my local network. 

I forward Organizr though dash.domain.com -> 443 to the outside. Everything works here as expected, certs load, etc. 

All works well with the tabs when the only information I populate from a URL perspective is my HTTPS subdomains.

In the remote location I use Organizr for however, I'm running into an issue where I can't get Organizr to recognize that I am in fact, remote. The problem I have in that location is there's a traffic-sniffing network that blocks any site without a cert, or anything that looks to be non-HTTPS. 

What I'm running into is that the moment I put a local URL or a ping URL into Organizr, (Which I want to be able to use when at home) , and attempt to use dash.domain.com remotely, the frame won't attempt the HTTPS URL anymore, it defaults to the local one, even though I'm accessing Organizr outside of my local network.

I can verify this, as if i switch from frames to new tab, it's loading 192.168.1.x into the new tab, instead of the URL of the HTTPS address. 

I can see in the debug menu in Organizr, that I have "local" set to true, even when accessing from outside my network. 

Was digging through threads, found this, where someone else mentioned specific Nginx Settings:

https://github.com/causefx/Organizr/issues/1263

 _"i was able to resolve this issue by making sure my nginx had the **x-forwarded-for** set to **$remote_addr** and now it shows the external on login as well as i can be on the internal ip when at home"_

When reconning how to do that via Nginx Proxy manager, that seems to be baked in via turning WebSocket support on which I have on for all of my forwarded subdomains, including the one I use for Organizr itself. 

https://github.com/NginxProxyManager/nginx-proxy-manager/issues/1556

"
![image](https://user-images.githubusercontent.com/90459948/168193336-711229dc-d66f-47fe-b66b-36d4d3b240b2.png)
"

Anyone have any experience with this, or have any pointers? Appreciate any assist.

**Comment by HalianElf** [comment]
Go to `<path to mapped /config>/nginx/site-confs/default` and uncomment the `set_real_ip_from` and make sure the subnet on that line matches your docker subnet

**Comment by dgarra** [comment]
Hey thanks for the response. 

Wasn't aware of that setting, gave it a shot, but sadly no go. 


![image](https://user-images.githubusercontent.com/90459948/168199488-4ad45f0e-c439-4041-b43d-98677bd44084.png)

Organizr still sees me as inside the network, when out. 

For the record, my main network is 192.168.0.0/24, tried that as well, same thing.

**Comment by HalianElf** [comment]
Can you open your settings and see what IP it's seeing you come from? Opening the settings page should trigger a Notice message in the logs that you can check it on.

**Comment by causefx** [comment]
yeah lets see what IP its seeing you as from the logs page in settings.

**Comment by dgarra** [comment]
Settings -> logs, all show 172.17.0.1. Seems to log my docker gateway IP for everything as these time stamps represent both internal and external activity. 

![image](https://user-images.githubusercontent.com/90459948/168203152-8833c418-682b-4d93-adfd-c3154bf6bc62.jpeg)

**Comment by HalianElf** [comment]
Subnet in the nginx config should be good then... did you restart the container after changing it to make it take effect?

**Comment by dgarra** [comment]
I did. Can confirm container restarts after modifying the default file.

**Comment by causefx** [comment]
are you able to create a php file inside org container and access it via domain?  create a file with this:

```
<?php

phpinfo();
```

**Comment by dgarra** [comment]
@causefx apologies, could you be a bit more verbose? 

Create a file with that info in the config directory of the organizr container, correct? 

No problem, dropped that on the root of the container dir, how can I access it via the domain as per your test? 

Apologies, PHP isn't a strong suit, appreciate any assist. 

![image](https://user-images.githubusercontent.com/90459948/168401729-be4b0b89-396a-418e-be5c-7db6a6d38a35.png)


<img width="443" alt="image" src="https://user-images.githubusercontent.com/90459948/168401717-1b5ea58a-7b9c-448a-a87f-f0b1d24b260c.png">

**Comment by dgarra** [comment]
@causefx, I follow to access this file in a browser, but is there specific dir in the container it needs to be in? Root doesn't get me anywhere, i don't see a "web" folder anywhere either. 

Thanks,

**Comment by causefx** [comment]
sorry for late response.  Put it in the folder where index.php is...  you will also see an api folder etc.

**Comment by dgarra** [comment]
@causefx , no problem at all, at your leisure, appreciate the help. 

File created: 
![Screen Shot 2022-05-20 at 12 33 29 PM](https://user-images.githubusercontent.com/90459948/169573249-3be9ca6c-be3d-4f80-bc70-b8237282deea.png)

![Screen Shot 2022-05-20 at 12 34 09 PM](https://user-images.githubusercontent.com/90459948/169573258-9ff4def6-f71d-4187-a00b-005f791f7b67.png)

Run over http local IP, doesn't seem to load:  

![Screen Shot 2022-05-20 at 12 34 42 PM](https://user-images.githubusercontent.com/90459948/169573340-bc526740-701f-4854-a196-757f61f370cf.png)

Run via HTTPS - domain, same result: 

![Screen Shot 2022-05-20 at 12 34 52 PM](https://user-images.githubusercontent.com/90459948/169573379-41a651e4-c1f1-49dd-8c80-36e6691fc839.png)

Can you confirm this is what you're looking for from the test? 

Thanks again,

**Comment by causefx** [comment]
put it inside that `organizr` directory then you will be able to access it

**Comment by dgarra** [comment]
@causefx , 

There we go. 

I can confirm it's accessible via both the localIP:port/test.php via my local network, 

**and** via the dash.xdomainx.com/test.php. from **outside** of my local network. 

Do you need an export of this diagnostic page? 


Thanks,

**Comment by causefx** [comment]
yes please, or email me the link so i can look at it

causefx at me dot com

**Comment by causefx** [comment]
as you can see from the info page - NPM is sending these values over:
![image](https://user-images.githubusercontent.com/16184466/169606015-f4d09118-1ed8-4f86-ac7f-fed49a3c7548.png)

**Comment by dgarra** [comment]
@causefx, 

Thanks for looking at it. 

This is odd, as no matter what config or test I make, I get the same results out of the test export, always in the docker network 172.17.0.0. 

1) I've changed the #set_real_IP_from value to 192.168.0.0/24 which is the subnet my docker host sits on, and regardless of if I test it internal or external, the test.php file still advertises the 172.17.0.1. 

2) This would line up to everything defaulting to the local IPs instead of the desired subdomain URLs, etc. 

Can you think of any test I could make to get the container to see a **different** value than 172.17.0.1, even if incorrect? 

The only tests I can think of is to change either/both my nginx or Organizr containers from bridged networking to macvlans, and see if I get different behavior. 

Thanks,

**Comment by causefx** [comment]
I tested adding organizr to NPM and didn't hit this issue at all...

Settings:
![image](https://user-images.githubusercontent.com/16184466/169657290-b37d5c58-34d3-41ef-bda8-85a04bd21631.png)
![image](https://user-images.githubusercontent.com/16184466/169657302-2939934f-ae63-4b87-a5c3-906b0fad32e9.png)

and nothing in advanced...

**Comment by dgarra** [comment]
So I have another NAS/docker host in a totally different network, in another location, so I basically ignored my current setup entirely to run additional tests. 

I deployed Organizr brand new, a new Nignx Proxy manager, and a Kuma uptime container to be referenced via the dash. 

Chose the same Nginx options as your above pics, just with my domain names. 

Host network here is 192.168.7.x, Docker network is 172.17.0.x. I have it pointing to a different domain/subdomain set name entirely. Containers are all set to bridged networking. 

I set the defaults file to remove the "#" comment, and set the real IP value to 172.17.0.0/16. 

Publicly, Organizr is exposed via Nginx via sub domain dash.newdomain.com

In Organizr, I've set the tab to 

Tab URL: kuma.newdomain.com
Tab Local URL: 192.168.7.10:3001
Ping URL : 192.168.7.10:3001

When I reach to dash.newdomain.com, organizr opens with a valid cert/https, no issues. 

When I use the Kuma tab, it opens via 192.168.7.10:3001 and errors out, as it's outside of the network. 

If I go back to the tab editor, and remove both the local IPs, only leaving kuma.newdomain.com, and open the tab again, it works fine, resolving as expected. 

debug menu in this setup gives the same, "local:true" when accessed publicly via dash.newdomain.com. 

TestPHP page looks the same. 

<img width="896" alt="image" src="https://user-images.githubusercontent.com/90459948/169670216-302a5c9c-e233-4053-b29e-77e1c70838c4.png">




So in summary, I took myself totally out of the main environment to rule to rule out everything, but unfortunately got the same behavior. 


The only similarity between these two setups is that the docker host is based on Synology DSM 6.2.x, which is what I'm questioning at this point, including the bridged networking of Docker itself. 

Any chance you run Synology/Docker in your setups?

**Comment by HalianElf** [comment]
Synology explains it https://www.pedrolamas.com/2020/11/04/exposing-the-client-ips-to-docker-containers-on-synology-nas/

**Comment by causefx** [comment]
ooohhhh try those commands. let us know.

**Comment by dgarra** [comment]
::::sigh:::::

Yep, that's totally it. Syno totally botches the stupid tables. 

![image](https://user-images.githubusercontent.com/90459948/169672270-cb97c822-9d42-4e25-8b60-2feb80937c6e.png)


![image](https://user-images.githubusercontent.com/90459948/169672225-2e383fc1-f40b-4d2c-86ab-4a9ffb5afb43.png)

<img width="911" alt="image" src="https://user-images.githubusercontent.com/90459948/169672261-a5e16cb8-7fa3-466e-a17d-95d715142cc7.png">

Seems I was somewhat on the right track with the MacVlan as the next step....I've seen issues with the bridged networking with PiHole a few years ago....specific to Synology now that I'm thinking about it. 

Well done @HalianElf . Well done indeed. 

@causefx I appreciate the time you took on this, nothing more annoying than subjecting environmental issues to developers. Above and beyond. Hopefully someone sees this thread and doesn't subject themselves to having to retest it. 

For the record, I've been using Organizr for the last few years, and it's by far the best dash for self hosting. Best of luck with it.

**Comment by dgarra** [comment]
If using a Synology Docker host, the IP tables are wrong, where client side IPs can't be exposed to the containers. 

This will prevent Organizr from observing if the calls are being made from outside the docker network. 

Added the following rules to the Synology/Docker host: 

sudo iptables -t nat -A PREROUTING -m addrtype --dst-type LOCAL -j DOCKER
sudo iptables -t nat -A PREROUTING -m addrtype --dst-type LOCAL ! --dst 127.0.0.0/8 -j DOCKER

Please note that a reboot of a syno OS will likely remove these. Please run these as a script that runs after reboot in Synology's task scheduler to put them back. 

Will open a case with Synology to complain about it, and update with results (if any....assuming unlikely) 

Please see for reference: 

https://www.pedrolamas.com/2020/11/04/exposing-the-client-ips-to-docker-containers-on-synology-nas/

**Comment by causefx** [comment]
place it in the same directory as index.php. then access it from your organizr domain root. mobile atm will type it out better once at computer.
